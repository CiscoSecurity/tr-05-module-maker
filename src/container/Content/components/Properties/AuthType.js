import React from "react";
import styles from "./Properties.module.scss";
import * as Constants from "../../../../globals/constants/constants";
import { toggleAuthType } from "../../../../redux/actions";
import { connect } from "react-redux";


class AuthType extends React.Component {
  constructor(props) {
        super(props);
        this.state = {
            auth_type_options: ["authorization-header", "basic", "bearer"],
        }
    }

render() {
    return (
        <div className={styles.divAuth}>

            <input type="checkbox"
                   autoComplete="off"
                   onClick={this.props.toggleAuthType}
            />

            <label>authorization</label>
            <select className={styles.selectAuth}
                    defaultValue={"DEFAULT"}
                    disabled={!Object.keys(this.props.syncProperties).includes("auth-type")}>
                <option value="DEFAULT" disabled hidden>{Constants.SELECT_PLACEHOLDER}</option>
                {
                   this.state.auth_type_options.map(
                        option => {
                            return <option>{option}</option>
                        }
                    )
                }
            </select>
        </div>
    )
}
}

const mapDispatchToProps = {
    toggleAuthType
}

const mapStateToProps = state => {
    return {
        syncProperties: state.json.properties
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(AuthType);
