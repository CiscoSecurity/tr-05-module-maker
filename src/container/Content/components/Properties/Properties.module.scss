.supportedAPI {
  padding-top: 3.75px;
  padding-bottom: 3.75px;
  align-content: center;
}

.divAuth {
  display: flex;
  align-items: center;
}

.selectAuth {
  flex: 1;
  margin-left: 4px;
}
