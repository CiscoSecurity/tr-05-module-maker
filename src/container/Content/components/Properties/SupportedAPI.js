import React from "react";
import styles from "./Properties.module.scss";
import { connect } from "react-redux";
import { addCapability, deleteCapability, updateSupportedAPI } from "../../../../redux/actions";


class SupportedAPI extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            title: this.props.api.title,
            class: this.props.api.class
        }
    }

    changeHandler = (event) => {
        this.props.updateSupportedAPI(this.state.title);

        if (event.target.checked) {
        this.props.addCapability(this.state.class);
        }
        else {
        this.props.deleteCapability(this.state.class);
        }
    }

    render() {
        return <div className={styles.supportedAPI}>
            <input type="checkbox" autoComplete="off" onChange={this.changeHandler}/>
            {this.state.title}
        </div>
    }
}

const mapDispatchToProps = {
    addCapability,
    deleteCapability,
    updateSupportedAPI
}

export default connect(null, mapDispatchToProps)(SupportedAPI);
