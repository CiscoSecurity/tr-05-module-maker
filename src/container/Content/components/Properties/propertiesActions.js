import {TOGGLE_AUTH_TYPE, UPDATE_AUTH_TYPE, UPDATE_SUPPORTED_API} from "../../../../globals/constants/types";

export function updateSupportedAPI(api) {
    return {
        type: UPDATE_SUPPORTED_API,
        payload: api
    }
}

export function toggleAuthType() {
    return {
        type: TOGGLE_AUTH_TYPE
    }
}

export function updateAuthType(type) {
    return {
        type: UPDATE_AUTH_TYPE,
        payload: type
    }
}
