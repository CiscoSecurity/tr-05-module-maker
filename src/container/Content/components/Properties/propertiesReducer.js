import {
    TOGGLE_AUTH_TYPE, UPDATE_AUTH_TYPE, UPDATE_CONF_TOKEN,
    UPDATE_SUPPORTED_API
} from "globals/constants/types"
import { TOGGLE_ALGORITHM } from "globals/constants/types";
import * as Constants from "globals/constants/constants";

const initialState = {
 "supported-apis": []
}

export const propertiesReducer = (state = initialState, action) => {
    switch (action.type) {
        case UPDATE_SUPPORTED_API:
            if (!state["supported-apis"].includes(action.payload)) {
                return {
                    ...state,
                        "supported-apis": state["supported-apis"].concat([action.payload])
                    }
            } else {
                return {
                    ...state,
                        "supported-apis": state["supported-apis"].filter(
                            (api) => api !== action.payload
                        )
                }
            }
        case TOGGLE_AUTH_TYPE:
            if (Object.keys(state).includes("auth-type")) {
                return Object.fromEntries(
                        Object.entries(state).filter(
                            ([key, val]) => key === "supported-apis"
                        )
                )
            } else {
                return {
                    ...state,
                        "auth-type": ""
                }
            }

        case UPDATE_AUTH_TYPE:
            return {
                ...state,
                "auth-type": action.payload
            }
        case UPDATE_CONF_TOKEN:
            return {...state,
                [action.payload.name]: action.payload.value
            }

        case TOGGLE_ALGORITHM:
            if (action.payload === 'RS256'){
                return Object.fromEntries(
                    Object.entries(state).filter(
                        ([key, val]) => key !== Constants.CONFIGURATION_TOKEN_KEY
                    )
                )
            }
            else {
                return Object.fromEntries(
                    Object.entries(state).filter(
                        ([key, val]) => key !== Constants.CUSTOM_JWKS_HOST
                    )
                )
            }

        default:
            return state
    }
}
