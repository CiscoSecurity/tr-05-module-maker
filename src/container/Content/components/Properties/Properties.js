import React from "react";
import SupportedAPI from "./components/SupportedAPI/SupportedAPI";
import AuthType from "./components/AuthType/AuthType";


class Properties extends React.Component {
    state = {
            apis: [
                {title: 'health', class: 'health'},
                {title: 'deliberate/observables', class: 'deliberate'},
                {title: 'refer/observables', class: 'refer'},
                {title: 'observe/observables', class: 'observe'},
                {title: 'respond/observables', class: 'respond'},
                {title: 'respond/trigger', class: 'respond'},
                {title: 'tiles', class: 'tiles'},
                {title: 'tiles/tile', class: 'tiles'},
                {title: 'tiles/tile-data', class: 'tiles'},
            ]
        }

    render() {
        return (
            <div>
                <p>Properties</p>
                <fieldset>
                    <p>Supported APIs</p>
                    <fieldset>
                        {
                            this.state.apis.map(
                                api => {
                                    return <SupportedAPI api={api} key={api.title}/>
                                }
                            )
                        }
                    </fieldset>
                    <p>Auth-type</p>
                    <fieldset>
                        <AuthType/>
                    </fieldset>
                </fieldset>
            </div>
        )
    }
}

export default Properties;
