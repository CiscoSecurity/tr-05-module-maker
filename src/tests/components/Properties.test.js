import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import Properties from "../../container/Content/components/Properties/Properties";
import React from "react";
import configureStore from 'redux-mock-store';
import toJson from "enzyme-to-json";


const mockStore = configureStore([]);


describe('Properties', () => {
    it('should render Properties with disabled AuthType select', () => {
        const store = mockStore({
            properties: {'supported-apis': []},
        });

        const wrapper = mount(
            <Provider store={store}>
                <Properties/>
            </Provider>);
        expect(wrapper.find('SupportedAPI')).toHaveLength(9);
        expect(wrapper.find('AuthType select').props()['disabled']).toBeTruthy();
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('should render Properties with enabled AuthType select', () => {
        const store = mockStore({
            properties: {
                'supported-apis': [],
                'auth-type': ''
            }
        });

        const wrapper = mount(
            <Provider store={store}>
                <Properties/>
            </Provider>);
        expect(wrapper.find('SupportedAPI')).toHaveLength(9);
        expect(wrapper.find('AuthType select').props()['disabled']).toBeFalsy();
        expect(toJson(wrapper)).toMatchSnapshot();
    })
});
