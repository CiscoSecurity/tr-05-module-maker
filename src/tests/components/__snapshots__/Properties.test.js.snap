// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties should render Properties with disabled AuthType select 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Properties>
    <div>
      <p>
        Properties
      </p>
      <fieldset>
        <p>
          Supported APIs
        </p>
        <fieldset>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "health",
                "title": "health",
              }
            }
            key="health"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "health",
                  "title": "health",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                health
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "deliberate",
                "title": "deliberate/observables",
              }
            }
            key="deliberate/observables"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "deliberate",
                  "title": "deliberate/observables",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                deliberate/observables
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "refer",
                "title": "refer/observables",
              }
            }
            key="refer/observables"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "refer",
                  "title": "refer/observables",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                refer/observables
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "observe",
                "title": "observe/observables",
              }
            }
            key="observe/observables"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "observe",
                  "title": "observe/observables",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                observe/observables
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "respond",
                "title": "respond/observables",
              }
            }
            key="respond/observables"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "respond",
                  "title": "respond/observables",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                respond/observables
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "respond",
                "title": "respond/trigger",
              }
            }
            key="respond/trigger"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "respond",
                  "title": "respond/trigger",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                respond/trigger
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "tiles",
                "title": "tiles",
              }
            }
            key="tiles"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "tiles",
                  "title": "tiles",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                tiles
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "tiles",
                "title": "tiles/tile",
              }
            }
            key="tiles/tile"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "tiles",
                  "title": "tiles/tile",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                tiles/tile
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "tiles",
                "title": "tiles/tile-data",
              }
            }
            key="tiles/tile-data"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "tiles",
                  "title": "tiles/tile-data",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                tiles/tile-data
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
        </fieldset>
        <p>
          Auth-type
        </p>
        <fieldset>
          <Connect(AuthType)>
            <AuthType
              syncProperties={
                Object {
                  "supported-apis": Array [],
                }
              }
              toggleAuthType={[Function]}
              updateAuthType={[Function]}
            >
              <div>
                <div
                  className="divAuth"
                >
                  <input
                    autoComplete="off"
                    checked={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <label>
                     
                    authorization
                     
                  </label>
                  <select
                    className="selectAuth"
                    disabled={true}
                    onChange={[Function]}
                    required={false}
                    value=""
                  >
                    <option
                      disabled={true}
                      hidden={true}
                      value=""
                    >
                      Select your option
                    </option>
                    <option
                      key="authorization-header"
                      value="authorization-header"
                    >
                      authorization-header
                    </option>
                    <option
                      key="basic"
                      value="basic"
                    >
                      basic
                    </option>
                    <option
                      key="bearer"
                      value="bearer"
                    >
                      bearer
                    </option>
                    <option
                      key="configuration-token"
                      value="configuration-token"
                    >
                      configuration-token
                    </option>
                  </select>
                </div>
              </div>
            </AuthType>
          </Connect(AuthType)>
        </fieldset>
      </fieldset>
    </div>
  </Properties>
</Provider>
`;

exports[`Properties should render Properties with enabled AuthType select 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Properties>
    <div>
      <p>
        Properties
      </p>
      <fieldset>
        <p>
          Supported APIs
        </p>
        <fieldset>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "health",
                "title": "health",
              }
            }
            key="health"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "health",
                  "title": "health",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                health
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "deliberate",
                "title": "deliberate/observables",
              }
            }
            key="deliberate/observables"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "deliberate",
                  "title": "deliberate/observables",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                deliberate/observables
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "refer",
                "title": "refer/observables",
              }
            }
            key="refer/observables"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "refer",
                  "title": "refer/observables",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                refer/observables
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "observe",
                "title": "observe/observables",
              }
            }
            key="observe/observables"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "observe",
                  "title": "observe/observables",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                observe/observables
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "respond",
                "title": "respond/observables",
              }
            }
            key="respond/observables"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "respond",
                  "title": "respond/observables",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                respond/observables
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "respond",
                "title": "respond/trigger",
              }
            }
            key="respond/trigger"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "respond",
                  "title": "respond/trigger",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                respond/trigger
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "tiles",
                "title": "tiles",
              }
            }
            key="tiles"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "tiles",
                  "title": "tiles",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                tiles
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "tiles",
                "title": "tiles/tile",
              }
            }
            key="tiles/tile"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "tiles",
                  "title": "tiles/tile",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                tiles/tile
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
          <Connect(SupportedAPI)
            api={
              Object {
                "class": "tiles",
                "title": "tiles/tile-data",
              }
            }
            key="tiles/tile-data"
          >
            <SupportedAPI
              addCapability={[Function]}
              api={
                Object {
                  "class": "tiles",
                  "title": "tiles/tile-data",
                }
              }
              deleteCapability={[Function]}
              syncSupportedAPIs={Array []}
              updateSupportedAPI={[Function]}
            >
              <div
                className="supportedAPI"
              >
                <input
                  autoComplete="off"
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
                tiles/tile-data
              </div>
            </SupportedAPI>
          </Connect(SupportedAPI)>
        </fieldset>
        <p>
          Auth-type
        </p>
        <fieldset>
          <Connect(AuthType)>
            <AuthType
              syncProperties={
                Object {
                  "auth-type": "",
                  "supported-apis": Array [],
                }
              }
              toggleAuthType={[Function]}
              updateAuthType={[Function]}
            >
              <div>
                <div
                  className="divAuth"
                >
                  <input
                    autoComplete="off"
                    checked={true}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <label>
                     
                    authorization
                     
                  </label>
                  <select
                    className="selectAuth"
                    disabled={false}
                    onChange={[Function]}
                    required="required"
                    value=""
                  >
                    <option
                      disabled={true}
                      hidden={true}
                      value=""
                    >
                      Select your option
                    </option>
                    <option
                      key="authorization-header"
                      value="authorization-header"
                    >
                      authorization-header
                    </option>
                    <option
                      key="basic"
                      value="basic"
                    >
                      basic
                    </option>
                    <option
                      key="bearer"
                      value="bearer"
                    >
                      bearer
                    </option>
                    <option
                      key="configuration-token"
                      value="configuration-token"
                    >
                      configuration-token
                    </option>
                  </select>
                </div>
              </div>
            </AuthType>
          </Connect(AuthType)>
        </fieldset>
      </fieldset>
    </div>
  </Properties>
</Provider>
`;
